lib/note-content-editor.tsx

```diff
@@ -170,7 +170,9 @@ class NoteContentEditor extends Component<Props> {
         });
       }
     }, SPEED_DELAY);
-    this.focusEditor();
+    if (document?.activeElement?.id !== 'search-field') {
+      this.focusEditor();
+    }
```

Comment:
there are many ways this can go wrong, far more than the one way it can go right.

we should maybe formalize the rules for changing focus and then see if a more elegant solution drops out. maybe there's some global (Redux) state value tracking what context we're in that changes when we `enter` or `blur` the search field…

maybe we only care about the initial load? that is, we should grab `this.focusEditor()` _unless_ the search bar has already been focused, something we could grab from Redux or from a window-global value - a latch that only turns on the first time the search field is focused.

Reply:
Okay, I will revisit. I agree that we should change the focus as little as possible. It's pretty annoying when the editor fails to get focus, but we really only want to force-focus the editor if there's not another input element focused (in practice this means the search field, although I suppose it might be interesting to keep focus on the tag bar through a note selection change).

That being said:

> there are many ways this can go wrong,

I'm curious if you found any bugs with this implementation or know of a case where checking the `activeElement` would fail.

Reply:
I'm not sure that we only care about initial load, as @belcherj pointed out, right now it also refocuses the editor after you change notes.

Reply:
it's not about `activeElement` as much as it's `!== 'search-field'` since it's the kind of thing that can end up feeling like spooky action at a distance when we start working with focus on something else, like sort direction or the tag list.

I didn't try this out and I didn't find or see an immediate error with it; just noting that this could be hard to find when the next PR dealing with focus comes around.

to my understanding, re-focusing the editor after changing notes has been intentional since you might want to be opening up a note and then start typing. they keyboard shortcuts still allow you to navigate the note list even when the focus is being taken.

is it only on boot or is it on every note change we care about preserving focus? is there some kind of even that "resets" focus and then the first time to get it keeps it? it sounds as we talk like we might be wanting a global focus latch with a first-come-first-keep rule about it. I'm definitely not sure of this of course.